### PR TITLE
Limit size of HeronTupleSet.

### LINKED ISSUE #2234: mempool growth issue
This issue is from Heron 0.14.11 (in which we are still using `HeronTupleSet`, unlike master branch).

A topology in Twitter environment has high memory usage. We found out its memory pool is growing without bound. #2073 assumes that each message in mempool is around 8MB. However, we observed one single message becomes extremely large. And it is strange that there is **only one** message of this type in the memory pool.

`N5heron5proto6system13HeronTupleSetE - nums: 1 - bytes used: 657693336`

We then printed out the size of each incoming `HeronTupleSet` in bytes. Here is the [increase](https://gist.github.com/objmagic/dafe983715fc1ae9c8c8cf6a75a664ec). It seems we are using this single one message repeated and although we clear it every time we acquire it, this protobuf message object keeps growing. 

cc @srkukarni @huijunw

### PR DESCRIPTION
**Pull request by objmagic**
Fix #2234. We limit the size of `HeronTupleSet`. If it is larger than the maximum size, we release it back to allocator instead of memory pool.

Tested on local machine.

### CODE REVIEW

File: heron/stmgr/src/cpp/manager/stmgr-server.cpp [hunk 1]
```diff
@@ -431,7 +433,15 @@ void StMgrServer::HandleTupleSetMessage(Connection* _conn,
         ->incr_by(_message->control().fails_size());
   }
   stmgr_->HandleInstanceData(iter->second, instance_info_[iter->second]->local_spout_, _message);
-  __global_protobuf_pool_release__(_message);
+  auto message_size = _message->ByteSize();
```
Comment (srkukarni) [inline heron/stmgr/src/cpp/manager/stmgr-server.cpp]: ByteSizeLong?

Comment (objmagic) [reply heron/stmgr/src/cpp/manager/stmgr-server.cpp]: I could not find `ByteSizeLong` API in generated code. Let me try again...

Comment (objmagic) [reply heron/stmgr/src/cpp/manager/stmgr-server.cpp]: seems this is [introduced in 3.1.0](https://github.com/google/protobuf/blob/master/CHANGES.txt#L319) and then [deprecated in 3.4.0](https://github.com/google/protobuf/blob/master/CHANGES.txt#L58)

Comment (objmagic) [reply heron/stmgr/src/cpp/manager/stmgr-server.cpp]: also, ByteSize calculates size of serialized message, which seems will be slightly larger than actual size in memory.

Comment (objmagic) [reply heron/stmgr/src/cpp/manager/stmgr-server.cpp]: so if the message is 60MB but only little part of it is used to hold the incoming message (because incoming message is small), we will not delete it because I believe serialized size is much smaller than 60MB.

**Comment by objmagic** [comment]
Did some experiment:

```
  LOG(INFO) << _message->ByteSize();
  LOG(INFO) << _message->SpaceUsed();
  _message->Clear();
  LOG(INFO) << _message->ByteSize();
  LOG(INFO) << _message->SpaceUsed();
```

gives us

```
I0828 14:10:23.306396 2502742976 stmgr-server.cpp:437] 97
I0828 14:10:23.306404 2502742976 stmgr-server.cpp:438] 832
I0828 14:10:23.306416 2502742976 stmgr-server.cpp:440] 0
I0828 14:10:23.306422 2502742976 stmgr-server.cpp:441] 832
```

**Comment by srkukarni** [comment]
Thats interesting obs. So ByteSize is probably more related to wire format size and SpaceUsed is related to actual in memory repr. In that case, shouldn't you use SpaceUsed? But isn;t that very slow?

**Comment by objmagic** [comment]
@srkukarni It seems that we have no choice but to use `SpaceUsed` here. We can do some benchmark here to measure its performance.

**Comment by huijunw** [comment]
If the performance is a concern, I would suggest the old method: put it in mempool and run a garage collection against mempool to remove large tuples every 1 min.

**Comment by objmagic** [comment]
Some benchmarking for `SpaceUsed`. It looks not bad @srkukarni 

```
I0828 15:22:59.759600 2502742976 stmgr-server.cpp:439] size: 1105717
I0828 15:22:59.759625 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:22:59.967779 2502742976 stmgr-server.cpp:439] size: 1250892
I0828 15:22:59.967838 2502742976 stmgr-server.cpp:440] time: 10 microseconds
I0828 15:23:00.128224 2502742976 stmgr-server.cpp:439] size: 1250892
I0828 15:23:00.128247 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:00.290628 2502742976 stmgr-server.cpp:439] size: 1250892
I0828 15:23:00.290647 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:00.485448 2502742976 stmgr-server.cpp:439] size: 2150892
I0828 15:23:00.485477 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:00.642523 2502742976 stmgr-server.cpp:439] size: 2150892
I0828 15:23:00.642571 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:00.818881 2502742976 stmgr-server.cpp:439] size: 2150892
I0828 15:23:00.818902 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:00.970676 2502742976 stmgr-server.cpp:439] size: 3100892
I0828 15:23:00.970695 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:01.183621 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:01.183640 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:01.350108 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:01.350129 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:01.512719 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:01.512740 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:01.721442 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:01.721462 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:01.896715 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:01.896742 2502742976 stmgr-server.cpp:440] time: 7 microseconds
I0828 15:23:02.010160 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:02.010188 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:02.057590 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:02.057623 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:02.272822 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:02.272848 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:02.437445 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:02.437464 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:02.597956 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:02.598012 2502742976 stmgr-server.cpp:440] time: 8 microseconds
I0828 15:23:02.817160 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:02.817188 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:03.026859 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:03.026887 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:03.198411 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:03.198441 2502742976 stmgr-server.cpp:440] time: 7 microseconds
I0828 15:23:03.354770 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:03.354794 2502742976 stmgr-server.cpp:440] time: 7 microseconds
I0828 15:23:03.476467 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:03.476487 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:03.522660 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:03.522689 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:03.737485 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:03.737577 2502742976 stmgr-server.cpp:440] time: 10 microseconds
I0828 15:23:03.908659 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:03.908843 2502742976 stmgr-server.cpp:440] time: 7 microseconds
I0828 15:23:04.066982 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:04.067008 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:04.236320 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:04.236377 2502742976 stmgr-server.cpp:440] time: 15 microseconds
I0828 15:23:04.403852 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:04.403882 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:04.557363 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:04.557399 2502742976 stmgr-server.cpp:440] time: 8 microseconds
I0828 15:23:04.720309 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:04.720326 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:04.892799 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:04.892851 2502742976 stmgr-server.cpp:440] time: 13 microseconds
I0828 15:23:05.048017 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:05.048043 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:05.164396 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:05.164415 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:05.210952 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:05.210978 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:05.377094 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:05.377117 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:05.595759 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:05.595778 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:05.753278 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:05.753295 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:05.962968 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:05.962990 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:06.130947 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:06.130966 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:06.291152 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:06.291170 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:06.462307 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:06.462333 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:06.626914 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:06.626945 2502742976 stmgr-server.cpp:440] time: 7 microseconds
I0828 15:23:06.834550 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:06.834570 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:07.042392 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:07.042430 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:07.254945 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:07.254994 2502742976 stmgr-server.cpp:440] time: 9 microseconds
I0828 15:23:07.423046 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:07.423089 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:07.586993 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:07.587013 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:07.753795 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:07.753813 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:07.911969 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:07.912003 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:07.924479 2502742976 tmaster-client.cpp:203] Sending heartbeat
I0828 15:23:08.082916 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:08.082938 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:08.239189 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:08.239215 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:08.355965 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:08.355984 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:08.402683 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:08.402707 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:08.566846 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:08.566872 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:08.736565 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:08.736585 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:08.945924 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:08.945942 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:09.154872 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:09.154897 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:09.317586 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:09.317611 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:09.489550 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:09.489574 2502742976 stmgr-server.cpp:440] time: 7 microseconds
I0828 15:23:09.643889 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:09.643908 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:09.856318 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:09.856396 2502742976 stmgr-server.cpp:440] time: 8 microseconds
I0828 15:23:10.019237 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:10.019256 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:10.181098 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:10.181120 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:10.342578 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:10.342607 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:10.503144 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:10.503161 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:10.676255 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:10.676275 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:10.841310 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:10.841331 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:10.999248 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:10.999272 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:11.162648 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:11.162667 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:11.371052 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:11.371076 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:11.537482 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:11.537502 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:11.743221 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:11.743244 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:11.956501 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:11.956526 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:12.168359 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:12.168383 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:12.333200 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:12.333220 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:12.492785 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:12.492820 2502742976 stmgr-server.cpp:440] time: 7 microseconds
I0828 15:23:12.652281 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:12.652299 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:12.862710 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:12.862733 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:13.038830 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:13.038893 2502742976 stmgr-server.cpp:440] time: 7 microseconds
I0828 15:23:13.244649 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:13.244680 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:13.412093 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:13.412122 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:13.582895 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:13.582919 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:13.742321 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:13.742344 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:13.955909 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:13.955935 2502742976 stmgr-server.cpp:440] time: 7 microseconds
I0828 15:23:14.114053 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:14.114079 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:14.288125 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:14.288142 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:14.495038 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:14.495064 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:14.658213 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:14.658237 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:14.822262 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:14.822279 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:14.985718 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:14.985741 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:15.148098 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:15.148121 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:15.356293 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:15.356310 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:15.519220 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:15.519242 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:15.723994 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:15.724021 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:15.935029 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:15.935065 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:16.098445 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:16.098469 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:16.267468 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:16.267493 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:16.470955 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:16.470985 2502742976 stmgr-server.cpp:440] time: 7 microseconds
I0828 15:23:16.637337 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:16.637365 2502742976 stmgr-server.cpp:440] time: 8 microseconds
I0828 15:23:16.798907 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:16.798941 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:16.963367 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:16.963388 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:17.120815 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:17.120851 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:17.280721 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:17.280755 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:17.446753 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:17.446771 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:17.620613 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:17.620645 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:17.817011 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:17.817024 2502742976 stmgr-server.cpp:440] time: 2 microseconds
I0828 15:23:17.924962 2502742976 tmaster-client.cpp:203] Sending heartbeat
I0828 15:23:18.022199 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:18.022236 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:18.187273 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:18.187294 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:18.349710 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:18.349728 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:18.514205 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:18.514226 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:18.723558 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:18.723593 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:18.891324 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:18.891341 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:19.102291 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:19.102329 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:19.316295 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:19.316313 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:19.524253 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:19.524273 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:19.689882 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:19.689901 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:19.859247 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:19.859266 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:20.066699 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:20.066730 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:20.229720 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:20.229738 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:20.400859 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:20.400878 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:20.602530 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:20.602548 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:20.814083 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:20.814102 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:20.972462 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:20.972493 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:21.187356 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:21.187376 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:21.398550 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:21.398567 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:21.564195 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:21.564215 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:21.778362 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:21.778399 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:21.937408 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:21.937433 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:22.058609 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:22.058665 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:22.103879 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:22.103900 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:22.271633 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:22.271654 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:22.323895 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:22.323915 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:22.540364 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:22.540382 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:22.706812 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:22.706830 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:22.866564 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:22.866588 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:23.028275 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:23.028300 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:23.189738 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:23.189771 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:23.407835 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:23.407855 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:23.564306 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:23.564368 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:23.778379 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:23.778398 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:23.981071 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:23.981096 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:24.146374 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:24.146394 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:24.358403 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:24.358422 2502742976 stmgr-server.cpp:440] time: 3 microseconds
I0828 15:23:24.516858 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:24.516882 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:24.678105 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:24.678205 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:24.848099 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:24.848124 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:24.973999 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:24.974026 2502742976 stmgr-server.cpp:440] time: 8 microseconds
I0828 15:23:25.019623 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:25.019644 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:25.224201 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:25.224230 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:25.439513 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:25.439535 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:25.656437 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:25.656468 2502742976 stmgr-server.cpp:440] time: 5 microseconds
I0828 15:23:25.773116 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:25.773136 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:25.823282 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:25.823310 2502742976 stmgr-server.cpp:440] time: 4 microseconds
I0828 15:23:26.032425 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:26.032450 2502742976 stmgr-server.cpp:440] time: 6 microseconds
I0828 15:23:26.243134 2502742976 stmgr-server.cpp:439] size: 4000892
I0828 15:23:26.243151 2502742976 stmgr-server.cpp:440] time: 3 microseconds
```

**Comment by srkukarni** [comment]
Could you also share the thruput figures? Particularly in exclamation or other topologies that are used for Heron benchmarking?

**Comment by objmagic** [comment]
word count, parallelism=20

Stmgr CPU user time doubled, Data Tuples from Instances dropped ~25%.

We need to fix https://github.com/twitter/heron/issues/1908 first to see more metrics.
